src/trio/_tests/type_tests/raisesgroup.py

```diff
@@ -134,3 +134,26 @@ def check_nested_raisesgroups_matches() -> None:
     # has the same problems as check_nested_raisesgroups_contextmanager
     if RaisesGroup(RaisesGroup(ValueError)).matches(exc):
         assert_type(exc, BaseExceptionGroup[RaisesGroup[ValueError]])
+
+
+def check_multiple_exceptions_1() -> None:
+    a = RaisesGroup(ValueError, ValueError)
+    b = RaisesGroup(Matcher(ValueError), Matcher(ValueError))
+    c = RaisesGroup(ValueError, Matcher(ValueError))
+
+    d: BaseExceptionGroup[ValueError]
+    d = a
+    d = b
+    d = c
+    assert d
+
+
+def check_multiple_exceptions_2() -> None:
+    # "Cannot infer argument 1 of RaisesGroup"
+    # if removing overloads from Matcher then pyright stops warning, but mypy still
+    # complains
```

Comment:
will you file an issue to these tools?

Reply:
yeah, see https://github.com/python-trio/trio/pull/2989#issuecomment-2057109799

Reply:
The issue was in fact just that the typevar wasn't specified as covariant.